cctrl/auth.py

```diff
@@ -27,7 +29,51 @@
 except ImportError:
     import simplejson as json
 
-from cctrl.error import messages, PasswordsDontMatchException
+from pycclib import cclib
+
+from cctrl.keyhelpers import get_default_ssh_key_path, \
+    get_public_key_fingerprint, sign_token
+from cctrl.error import SignatureException, \
+    PublicKeyException, messages, PasswordsDontMatchException
+
+
+def create_config_dir(settings):
+    if os.path.isdir(settings.home_path):
+        return
+    elif os.path.isfile(settings.home_path):
+        print >> sys.stderr, 'Error: ' + settings.home_path + ' is a file, not a directory.'
+        sys.exit(1)
+    else:
+        os.mkdir(settings.home_path)
+
+
+def create_token(api, settings, email, password):
+    while True:
+        try:
+            if get_configfile(settings).get('ssh_auth') and not password:
+                key_path = get_configfile(settings).get('ssh_path',
+                                                        get_default_ssh_key_path())
+                fingerprint = get_public_key_fingerprint(key_path)
+                if not fingerprint:
+                    raise PublicKeyException('WrongPublicKey')
+
+                ssh_token = api.create_ssh_token()
+                signature = sign_token(key_path, fingerprint, ssh_token)
+                return api.create_token_ssh_auth(email,
+                                                 ssh_token,
+                                                 signature,
+                                                 fingerprint)
+            else:
+                return api.create_token_basic_auth(email, password)
+
+        except (cclib.APIException, SignatureException, PublicKeyException) as e:
+            if password:
```

Comment:
after taking a second look, it makes sense to change it (we don't reject empty passwords on pwd prompt).